presto-spi/src/main/java/com/facebook/presto/spi/security/SystemAccessControl.java

```diff
@@ -301,4 +302,22 @@ default void checkCanRevokeTablePrivilege(Identity identity, Privilege privilege
     {
         denyRevokeTablePrivilege(privilege.toString(), table.toString());
     }
+
+    /**
+     * Check if identity is allowed to show roles on the specified catalog.
+     *
+     * @throws com.facebook.presto.spi.security.AccessDeniedException if not allowed
+     */
+    default void checkCanShowRoles(Identity identity, String catalogName)
+    {
+        denyShowRoles(catalogName);
+    }
+
+    /**
+     * Filter the list of roles to those visible to the identity in the given catalog.
+     */
+    default Set<String> filterRoles(Identity identity, String catalogName, Set<String> roles)
```

Comment:
dropped in https://github.com/prestosql/presto/pull/90/commits/e2a2ff03d78734065ef0165225cebbb9dc3af89f